### ISSUE TITLE
Very weird problem - suspected caching problem with groups

### ISSUE THREAD

**Issue by knubew** [pods-framework/pods#6971]
Hello,

I've had a very strange problem with Pods for a long time. If I've observed it correctly, the problem always appears after a pods update. I suspect it's a caching issue within pods, the "Pods Groups" because it also occurs when I disable my "W3 Total Cache" and "Autoptimize" plugins. Since I'm German, it's difficult for me to describe the problem exactly, but I'll try. I have attached various screenshots to help.

**What do I use Pods for?**
I use pods for different types of events from different organizers. Each organizer has the "Author" role and can edit their own events in their own frontend ("Avada Branding" plugin).

- The event data is organized in 4 different pods groups.
- One of the 4 groups is only visible to the admin role ("Restrict access by Role -> Administrator").
- The other 3 groups are visible to all authors and the administrator ("Restrict access by Role -> Author" / "Restrict access by Role -> Administrator" - in this order!)

**This is how everything looks WITHOUT errors (before a pods update):**

**01_ADMIN-View__Edit_Pods__BEFORE_Pods_update**
![01_ADMIN-View__Edit_Pods__BEFORE_Pods_update](https://user-images.githubusercontent.com/6688340/207591704-69e4fa99-fb72-44f7-8880-234affc833c5.jpg)

**02_ADMIN-View__Event_with_groups__BEFORE_Pods_update**
![02_ADMIN-View__Event_with_groups__BEFORE_Pods_update](https://user-images.githubusercontent.com/6688340/207591710-7974425d-c747-41fd-bc3e-bfa0834861ba.jpg) 

**03_AUTHOR-View__Edit_Pods__BEFORE_Pods_update**
![03_AUTHOR-View__Edit_Pods__BEFORE_Pods_update](https://user-images.githubusercontent.com/6688340/207591717-2c5c10b3-ea98-4ae3-b953-4f950dbd9344.jpg)

**04_AUTHOR-View__Event_with_groups__BEFORE_Pods_update**
![04_AUTHOR-View__Event_with_groups__BEFORE_Pods_update](https://user-images.githubusercontent.com/6688340/207591719-27b679e4-54eb-4988-a3fe-b14739659779.jpg)

**Then I successfully do the pods update as admin:**

**05_ADMIN-View__Updates__BEFORE_Pods_update**
![05_ADMIN-View__Updates__BEFORE_Pods_update](https://user-images.githubusercontent.com/6688340/207591724-8178c1b1-9269-4d52-b056-6d8ad090f2c4.jpg)

**06_ADMIN-View__Updates__AFTER_Pods_update**
![06_ADMIN-View__Updates__AFTER_Pods_update](https://user-images.githubusercontent.com/6688340/207591727-650fe0f3-fa0b-4d4d-96a4-269261ccea83.jpg)

Then I log out (as admin) and log in as author with the login data of an organizer and call up an event ("Veranstaltung") for editing. **However, NO groups are displayed.**

**07_AUTHOR-View__Event_with_groups_ERROR__AFTER_Pods_update**
![07_AUTHOR-View__Event_with_groups_ERROR__AFTER_Pods_update](https://user-images.githubusercontent.com/6688340/207591730-a0cd7a7c-3beb-48c0-813c-5bbd3c8331b1.jpg)

Then I log out (as author) and log in again as "admin" and call up an event for editing as admin. As an admin, however, NO groups are displayed to me either.

**08_ADMIN-View__Event_with_groups_ERROR__AFTER_Pods_update**
![08_ADMIN-View__Event_with_groups_ERROR__AFTER_Pods_update](https://user-images.githubusercontent.com/6688340/207591740-bf898819-cd72-43e4-855f-a942315b6711.jpg)

I then just clear the pods cache to try to fix the problem.

**09_ADMIN-View__Clear_Pods_Cache__AFTER_Pods_update**
![09_ADMIN-View__Clear_Pods_Cache__AFTER_Pods_update](https://user-images.githubusercontent.com/6688340/207591743-df44a115-a016-44dd-84ad-7b1bd250d3d0.jpg)

Then I log out (as admin) and log in as "author" again with the login data of an organizer and call up an event ("Veranstaltung") for editing but the same error persists and NO pods groups are showing. Very, very weired...

**10_AUTHOR-View__Event_with_groups_ERROR__AFTER_Pods_cache_clear**
![10_AUTHOR-View__Event_with_groups_ERROR__AFTER_Pods_cache_clear](https://user-images.githubusercontent.com/6688340/207591747-a9955e47-c58e-4e30-987f-964f07c64ce2.jpg)

Then I log out (as author) and log in again as "admin" to clear the Pods cache again.

**BUT THIS TIME I CALL (as Admin) AN EVENT AFTER CLEARING THE PODS CACHE !**
Wow, the pods groups are displayed again without errors for me as an admin.

**11_ADMIN-View__Event_with_groups__OKAY_AFTER_Pods_cache_clear_and_Pods_call**
![11_ADMIN-View__Event_with_groups__OKAY_AFTER_Pods_cache_clear_and_Pods_call](https://user-images.githubusercontent.com/6688340/207591749-3e275775-ca85-4b9f-ac8d-c4ba110aa33a.jpg)

Now I check whether the author role shows the pods groups again and log out as admin and log in again as organizer (as author). Great, everything works as it should.

**12_AUTHOR-View__Event_with_groups__OKAY_AFTER_Pods_cache_clear_and_Pods_call**
![12_AUTHOR-View__Event_with_groups__OKAY_AFTER_Pods_cache_clear_and_Pods_call](https://user-images.githubusercontent.com/6688340/207591754-1ff110ca-8634-4538-9149-92553302ccb0.jpg)

**MY CONCLUSION:**
**Every time I update Pods, I have to do the following things AS ADMIN afterwards:**

**- Clear the Pods cache under Settings.
- AS ADMIN call up any event for editing and then close it again.**

Please excuse the long error description. It took me a long time to figure out how to work around the error that occurred. At least that's not logical for me, but I hope that the description will help you with troubleshooting.

**Comment by sc0ttkclark** [comment]
Thanks for the comprehensive bug report here @knubew, I'll look into this to see what I can find out. This definitely isn't normal and isn't something I've seen before so I'm very curious to find the solution for it.

Can you provide a package of your configuration (Pods Admin > Migrate Packages) using the Migrate Packages component (Pods Admin > Components)? That might help us to see if we can reproduce the issue.

Also helpful would be access to your Site Health information (Tools > Site Health > Info Tab > Copy site info to clipboard) would also help us determine if there are any potential conflicts from other plugins or your theme.

**Comment by knubew** [comment]
> Can you provide a package of your configuration (Pods Admin > Migrate Packages) using the Migrate Packages component (Pods Admin > Components)? That might help us to see if we can reproduce the issue.

Do you need the export with all templates?
Actually there are 22 templates in use.

Where can I send the export file to?
I don't want to post it publicly.

> Also helpful would be access to your Site Health information (Tools > Site Health > Info Tab > Copy site info to clipboard) would also help us determine if there are any potential conflicts from other plugins or your theme.

Where can I send the export file to?
I don't want to post it publicly.

**Comment by knubew** [comment]
I just sent you an e-mail with the requested data (in the e-mail attachment).

**Comment by sc0ttkclark** [comment]
@knubew Can you try the repair tool at Pods Admin > Settings > Tools > and run the "Repair Pod, Groups, and Fields" tool for that pod?

**Comment by knubew** [comment]
@sc0ttkclark I did this for every pod. The output message "No actions were needed." was always the same.

**Comment by sc0ttkclark** [comment]
Thanks for that, figured I'd ask just in case since I didn't ask before.

I'm going to look back over what you sent previously if I can manage to find it in my email chaos :)

**Comment by knubew** [comment]
I've just resend it to scott@skc.dev

**Comment by sc0ttkclark** [comment]
I've been digging heavily into this for weeks and believe I have a fix in Pods 2.9.12 which you can test out here: https://github.com/pods-framework/pods/archive/refs/heads/release/2.9.12.zip

Let me know if that version resolves the issue you've been seeing.

**Comment by knubew** [comment]
Hello Scott,

thank you for your dedicated work on this problem.

Unfortunately, I'll be traveling on business until April 2nd and unfortunately I don't have the opportunity to test the fix beforehand. But I'll be happy to do that when I get back.

**Comment by knubew** [comment]
Hello Scott,

after traveling on business until April 2nd, I would like to get in touch again regarding the problem I described.

I've been using the pods version 2.9.13 since April 10th, 2023 and the problem I described hasn't occurred again. You seem to have solved the problem. Thank you very much.

Nonetheless, I'll keep an eye on it for the foreseeable future. If the problem comes up again, I'll let you know.
